sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuExec.scala

```diff
@@ -103,25 +103,14 @@ object GpuMetric extends Logging {
   val DESCRIPTION_READ_FS_TIME = "time to read fs data"
   val DESCRIPTION_WRITE_BUFFER_TIME = "time to write data to buffer"
 
-  def unwrap(input: GpuMetric): SQLMetric = input match {
-    case w :WrappedGpuMetric => w.sqlMetric
-    case i => throw new IllegalArgumentException(s"found unsupported GpuMetric ${i.getClass}")
+  def unwrap(input: Map[String, GpuMetric]): Map[String, SQLMetric] = input.collect {
+    // remove the metrics that are not registered (NoopMetric)
+    case (k, w: WrappedGpuMetric) => (k, w.sqlMetric)
   }
 
-  def unwrap(input: Map[String, GpuMetric]): Map[String, SQLMetric] = input.filter {
-    // remove the metrics that are not registered
-    case (_, NoopMetric) => false
-    case _ => true
-    // sadly mapValues produces a non-serializable result, so we have to hack it a bit to force
-    // it to be materialized
-  }.mapValues(unwrap).toArray.toMap
-
-  def wrap(input: SQLMetric): GpuMetric = WrappedGpuMetric(input)
-
-  def wrap(input: Map[String, SQLMetric]): Map[String, GpuMetric] =
-  // sadly mapValues produces a non-serializable result, so we have to hack it a bit to force
-  // it to be materialized
-    input.mapValues(wrap).toArray.toMap
+  def wrap(input: Map[String, SQLMetric]): Map[String, GpuMetric] = input.map {
+    case (k, v) => (k, WrappedGpuMetric(v))
```

Comment:
Not related to this PR, but I find it today that the scala compiler can't figure out total function and partial function. So if I change the function in `map` as
```
case ("aaa", v) => ("aaa", WrappedGpuMetric(v))
```

The code can be compiled, but will return `matchError` at runtime. 
Not sure if it is a known feature (or bug) in the scala compiler.